src/provider.coffee

```diff
@@ -6,7 +6,7 @@ extensions        = require './extensions'
 
 
 class Provider
-  constructor: (consumer_key, consumer_secret, nonceStore, signature_method=(new HMAC_SHA1()) ) ->
+  constructor: ({consumer_key, consumer_secret, nonceStore, signature_method=(new HMAC_SHA1()), providerOptions}={}) ->
```

Comment:
If you are going to change the input to be an object, I would just add `authEnabled` to the object instead of nesting an options object within the parameters object.

```
constructor: ({consumer_key, consumer_secret, nonceStore, signature_method=(new HMAC_SHA1()), authEnabled=true})
``` 

Changing the parameters this way will break existing users of the library, so if this ever makes it into the "official" library we'll need to bump the version accordingly.